models/refresh_token.go

```diff
@@ -126,7 +127,14 @@ func createRefreshToken(tx *storage.Connection, user *User, oldToken *RefreshTok
 
 	if token.SessionId == nil {
 		// Existing refresh tokens may have a null session_id if they were created before v2.15.3
-		session, err := CreateSession(tx, user, params.FactorID)
+		var session *Session
+		var err error
+		// TODO(Joel): Find better workaround
+		if os.Getenv("MFA_ENABLED") == "true" {
```

Comment:
Maybe split this function in two, one calling MFA and the other not?

Reply:
Oh I think all the changes would have to bubble up though (e.g. we'd need to duplicate all the instances where `createRefreshToken` is called)

If it's okay with everyone I'd much prefer to stick with this for now in order to minimise the number of duplications we have. This should work but isn't too elegant